salt/modules/file.py

```diff
@@ -2361,7 +2361,7 @@ def blockreplace(path,
     try:
         fi_file = fileinput.input(path,
                     inplace=False, backup=False,
-                    bufsize=1, mode='r')
+                    bufsize=1, mode='rb')
```

Comment:
Again, I'm not sure what the reason is for this. This breaks almost everything below on Python 3. You're going to be reading in bytestrings and then trying to compare and do membership checks using strings. So below, when you check for the marker, that's going to blow up:

```
Python 3.6.2 (default, Jul 20 2017, 03:52:27)
[GCC 7.1.1 20170630] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fileinput
>>> fi_file = fileinput.input('foo.txt', inplace=False, backup=False, bufsize=1, mode='rb')
>>> line = fi_file.readline()
>>> line
b'Line one\n'
>>> marker_start = '#-- start managed zone --'
>>> marker_start in line
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
>>> fi_file.close()
```